localstack-core/localstack/plugins.py

```diff
@@ -21,3 +26,17 @@ def delete_cached_certificate():
     LOG.debug("Removing the cached local SSL certificate")
     target_file = get_cert_pem_file_path()
     rm_rf(target_file)
+
+
+class OASPlugin(Plugin):
+    namespace = "localstack.openapi.spec"
+
+    def __init__(self, spec_path: os.PathLike | str) -> None:
+        if isinstance(spec_path, str):
+            spec_path = Path(spec_path)
+        self.spec_path = spec_path
+        self.spec = {}
+
+    def load(self):
+        with self.spec_path.open("r") as f:
+            self.spec = yaml.safe_load(f)
```

Comment:
I think this plugin spec here is a bit confusing. It is only used in the validator for now, so I think the plugin definition should be there. The _registration_ of the _specific plugin_ (in this case the `CoreOASPlugin`) could be here though, as it's the `plugins.py` next to the actual spec (but could also stay where it is right now).